core-skills/web-application-development-with-react/README.md

```diff
@@ -0,0 +1,72 @@
+# Web Application Development with React
+
+## Summary
+
+1. Can describe situations where you should/shouldn't decide to use React for a project
+2. Can setup a React project from scratch in a repeatable way
+3. Can describe the different parts of the Javascript ecosystem (e.g. NPM, Webpack, etc)
+4. Can demonstrate using TDD on business logic within a React code base
+5. Can demonstrate strategies to achieve clear separation of business logic and display logic
+with React components
+6. Can demonstrate applying sensible styling to React components
+7. Can demonstrate strategies and tools to run tests against an API within a React application
+
+## Sections
+
+1. [React - Why and when should I use it?](./why-and-when.md)
+2. [Setting up a React project](./setup.md)
+3. [React Fundamentals](./fundamentals/README.md)
+4. [Prototyping React Components](./prototyping.md)
+5. [Styling React Components](./prototyping.md)
+
+## Mark Scheme
+
+### Hedgehog
+
+Learning the fundamentals - How not to hurt yourself with React
+
+- [ ] Can demonstrate an understanding of the benefits of using React
+- [ ] Creates an application with the following:
+  - [ ] The application can be run on any machine (e.g. Using Docker)
```

Comment:
An odd one in my opinion, are we explicitly asking for Docker support? What was the intent behind this one?

Reply:
From memory the intent was that we wanted to make sure people knew how to set up a React project that could be worked on by multiple people. Potentially you could know how to create one but it was tied to some globally installed package on your machine so no one else could pick it up. 

It does seem a bit strange to have it in here though on second reading :thinking: It depends on whether or not we feel the ability to set up a project to run anywhere is a general requirement of all our work and shouldn't be tested here? Hmm

Reply:
How else are you going to get `make serve` to just work on any machine that has docker installed

Reply:
In that case I’d reword it as:


The application has `make` recipies for `serve` and `test` which only depend on the host machine having Docker installed.